### PR TITLE
Add Xproj and project.json based version of Adventure example

### PR DESCRIPTION
**Pull request by TsengSR**
As requested in https://github.com/dotnet/orleans/issues/1083 here's a dotnet-cli based version of the Adventure sample. 

It's mainly unchanged except for adding project.json and xproj files as well as changing the `System.Web.Script.Serialization.JavaScriptSerializer` to `Json.NET` which is also used in ASP.NET Core as `System.Web.Script.Serialization.JavaScriptSerializer` wasn't ported to .NET Core.

I had to create a new folder for it. Attempting to add the json/xproj files side-by-side in the existing Adventure folder resulted in being unable to restore nuget packages when opening the old cproj based solution.

### CODE REVIEW

**Comment by sergeybykov** [comment]
This change puts a copy of the source files from Samples/Adventure to Samples/AdventureDotNetCli, which is obvious code duplication. Can we instead have the two sets of solution/project files in separate folders, but have them point to a single set of .cs and other files?

**Comment by TsengSR** [comment]
One could maybe put the *.csproj files outside of their respective folders and put the *.xproj/project.json inside the project folders. Doesn't work other way around as *.xproj/project.json don't have the option to add/remove a single file from solution. 

One can exclude them for compilation and publishing via `project.json`, Gonna try that when I am back at home

**Comment by TsengSR** [comment]
I've removed the *.cs files from the Adventure folder and referenced them from the AdventureDotNet folder. 

Still feels a bit "dirty", but there don't seem to be any other way. There are some workarounds to prevent VS restoring packages from `packages.xml` and only use `project.json`, but this would likely break support for VS 2013 where the .NET Core tools aren't available.

File: Samples/AdventureDotNetCli/AdventureGrainInterfaces/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Reflection;
```
Comment (shayhatsor) [inline Samples/AdventureDotNetCli/AdventureGrainInterfaces/Properties/AssemblyInfo.cs]: There's no need for AssemblyInfo.cs files in .net core projects.

**Comment by TsengSR** [comment]
Removed the AssemblyInfo.cs files from th DotNetCli project

**Comment by sergeybykov** [comment]
Thank you @TsengSR!

**Pull request merged**